app/scripts/model/dataProxy.js

```diff
@@ -340,28 +357,39 @@
             _osSurvivalAttrMeta.display_name = 'Overall Survival';
             _osSurvivalAttrMeta.filter = [];
             _osSurvivalAttrMeta.show = true;
-            _patientAttributes.unshift(_osSurvivalAttrMeta);
+            _osSurvivalAttrMeta.priority = 1;
+            _patientAttributes[_osSurvivalAttrMeta.attr_id] = _osSurvivalAttrMeta;
           }
 
           // add Cancer Study
           if (self.getCancerStudyIds().length > 1) {
-            _patientAttributes.unshift({
+            _patientAttributes['study_id'] = {
               "datatype": "STRING",
               "description": "Cancer Studies",
               "display_name": "Cancer Studies",
               "attr_id": "study_id",
               "view_type": "pie_chart",
               "filter": [],
               "show": true
-            });
+            };
           }
-
+          _.each(iViz.util.sortClinicalAttrs(_.values(_.extend({}, _patientAttributes, _sampleAttributes))), function(attr, index) {
+            var attrId = attr.attr_id;
+            if (attr.priority === -1) {
+              if (_patientAttributes.hasOwnProperty(attrId)) {
+                _patientAttributes[attrId].priority = 10 + index;
+              } else {
+                _sampleAttributes[attrId].priority = 10 + index;
+              }
+            }
+          });
+          
           _result.groups = {};
           _result.groups.patient = {};
           _result.groups.sample = {};
           _result.groups.group_mapping = {};
-          _result.groups.patient.attr_meta = _patientAttributes;
-          _result.groups.sample.attr_meta = _sampleAttributes;
+          _result.groups.patient.attr_meta = iViz.util.sortClinicalAttrs(_.values(_patientAttributes));
```

Comment:
Instead of applying 'sortClinicalAttrs' again why don't we use 'priority' attribute in each meta attribute to sort them?